dds_web/api/project.py

```diff
@@ -196,20 +146,166 @@ def post(self):
+                )
+        else:  # current status is in progress
+            if project.has_been_available:
+                # No change in deadline if made available before
+                deadline = project.current_deadline
+            else:
+                project.released = current_time
+
+        # Create row in ProjectStatuses
+        return models.ProjectStatuses(
+            status="Available", date_created=current_time, deadline=deadline
+        )
+
+    def retract_project(self, project: models.Project, current_time: datetime.datetime):
+        """Retract project: Make status In Progress.
+
+        Only possible from Available.
+        """
+        # Check if valid status transition
+        self.check_transition_possible(
+            current_status=project.current_status, new_status="In Progress"
+        )
+
+        return models.ProjectStatuses(status="In Progress", date_created=current_time)
+
+    def expire_project(
+        self, project: models.Project, current_time: datetime.datetime, deadline_in: int
+    ) -> models.ProjectStatuses:
+        """Expire project: Make status Expired.
+
+        Only possible from Available.
+        """
+        # Check if valid status transition
+        self.check_transition_possible(current_status=project.current_status, new_status="Expired")
+
+        if deadline_in > 30:
+            raise DDSArgumentError(
+                message="The deadline needs to be less than (or equal to) 30 days."
+            )
+
+        deadline = dds_web.utils.current_time(to_midnight=True) + datetime.timedelta(
+            days=deadline_in
+        )
+        return models.ProjectStatuses(
+            status="Expired", date_created=current_time, deadline=deadline
+        )
+
+    def delete_project(self, project: models.Project, current_time: datetime.datetime):
+        """Delete project: Make status Deleted.
+
+        Only possible from In Progress.
+        """
+        # Check if valid status transition
+        self.check_transition_possible(current_status=project.current_status, new_status="Deleted")
+
+        # Can only be Deleted if never made Available
+        if project.has_been_available:
+            raise DDSArgumentError(
+                "Project cannot be deleted if it has ever been made available, abort it instead."
+            )
+        project.is_active = False
+
+        # Deletes files (also commits session in the function - possibly refactor later)
+        RemoveContents().delete_project_contents(project=project)
+        self.rm_project_user_keys(project=project)
+
+        # Delete metadata from project row
+        self.delete_project_info(proj=project)
+        delete_message = (
+            f"\nAll files in project '{project.public_id}' deleted and project info cleared"
+        )
+
+        return models.ProjectStatuses(status="Deleted", date_created=current_time), delete_message
+
+    def archive_project(
```

Comment:
We can archive a project directly from 'In Progress' without the `is_aborted` flag ever being set. Is that what we wish to happen?

Reply:
Check again, I readded the check